### PR TITLE
feat: updates on profile page

### LINKED ISSUE #54: Updates on Profile Page
## Profile page improvements

- [x] Profile page theme to Primer
- [x] Separate routes for view and edit
- [x] Display overview of pocketbooks on profile page
- [x] UID is irrelevant, copy email instead

EDIT:
- [ ] Design edit profile UI
   - Inspiration from https://github.com/usamazansari/expenses-tracker/settings

### PR DESCRIPTION
**Pull request by usamazansari**
Closes #54 

### Developer Checklist (Definition of Done)

*Issue*

- [x] All acceptance criteria from the issue are met
- [x] Tested in latest Chrome/Firefox

*Code*

- [x] Branch is up-to-date with the branch to be merged with, i.e., develop
- [x] Code is cleaned up and formatted

*PR*

- [x] Descriptive title for this pull request is provided (will be used for release notes later)
- [x] Reviewer and assignees are defined
- [x] Add type label (e.g., bug, feature) to this pull request
- [x] Add release label (e.g., `release: minor`) to this PR following [semver](https://semver.org/)
- [x] The PR is connected to the corresponding issue (via `Closes #...`)
- [x] [Summary of changes](#summary-of-changes) is written

### Summary of changes
- Update Profile page of the app

### Screenshots
[62-profile-page-updates.webm](https://github.com/usamazansari/expenses-tracker/assets/98811057/b9151865-4518-4e1c-a3e7-c4c32b211b77)

### Additional notes for the reviewer(s)
- N/A

### CODE REVIEW

File: libs/features/profile/src/lib/components/profile-edit/profile-edit-password/profile-edit-password.component.scss
Comment (usama251993) [inline libs/features/profile/src/lib/components/profile-edit/profile-edit-password/profile-edit-password.component.scss:1]: Delete this file

File: libs/features/profile/src/lib/components/profile-edit/profile-edit.service.ts [hunk 1]
```diff
@@ -1,15 +1,19 @@
...
 
 import { AuthService, ContextService, ErrorService } from '@expenses-tracker/core';
 import { NotificationService, RoutePaths } from '@expenses-tracker/shared/common';
 import { IFlag } from '@expenses-tracker/shared/interfaces';
 
+export type EditMode = 'displayName' | 'password';
+
 export type ComponentFlags = {
-  edit: IFlag;
+  edit: { displayName: IFlag; password: IFlag };
```
Comment (usama251993) [inline libs/features/profile/src/lib/components/profile-edit/profile-edit.service.ts:13]: Use flags in the component

File: libs/shared/common/src/lib/utilities/form.utils.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import { AbstractControl, FormGroup } from '@angular/forms';
+import { FormControlExtras, FormGroupTypeGenerator } from '@expenses-tracker/shared/interfaces';
+
+import { camelCaseToSentenceCase } from './string.utils';
+
+const getError = (control: AbstractControl<unknown, unknown>): boolean => {
```
Comment (usama251993) [inline libs/shared/common/src/lib/utilities/form.utils.ts:6]: Add documentation

File: libs/shared/common/src/lib/utilities/form.utils.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import { AbstractControl, FormGroup } from '@angular/forms';
+import { FormControlExtras, FormGroupTypeGenerator } from '@expenses-tracker/shared/interfaces';
+
+import { camelCaseToSentenceCase } from './string.utils';
+
+const getError = (control: AbstractControl<unknown, unknown>): boolean => {
+  return control.touched && !!control.errors;
+};
+
+const getErrorMessage = <C extends { [key: string]: unknown }, F extends FormGroup<FormGroupTypeGenerator<C>>>(
```
Comment (usama251993) [inline libs/shared/common/src/lib/utilities/form.utils.ts:10]: Add documentation

File: libs/shared/common/src/lib/utilities/form.utils.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  if (formGroup['controls'][formControlName]?.hasError('email')) {
+    return `Invalid ${control.toLowerCase()} address`;
+  }
+  if (formGroup['controls'][formControlName]?.hasError('required')) {
+    return `${control} is required`;
+  }
+  return `Unknown validation error for ${control}`;
+};
+
+const controlStateValidator = <C extends { [key: string]: unknown }, F extends FormGroup<FormGroupTypeGenerator<C>>>(
```
Comment (usama251993) [inline libs/shared/common/src/lib/utilities/form.utils.ts:24]: Add documentation

File: libs/shared/common/src/lib/utilities/string.utils.ts [hunk 5]
```diff
@@ -0,0 +1,7 @@
+const camelCaseToSentenceCase = (input: string) =>
```
Comment (usama251993) [inline libs/shared/common/src/lib/utilities/string.utils.ts:1]: Add documentation

**Review by usama251993 — CHANGES_REQUESTED** [review]
Please address feedback

File: libs/features/profile/src/lib/components/profile-edit/profile-edit-password/profile-edit-password.component.scss
Comment (usamazansari) [reply libs/features/profile/src/lib/components/profile-edit/profile-edit-password/profile-edit-password.component.scss:1]: Resolved in https://github.com/usamazansari/expenses-tracker/pull/62/commits/6faffe4c5504e5a788c369bb48a0ac2d54eaa526

File: libs/features/profile/src/lib/components/profile-edit/profile-edit.service.ts (hunk 1 above)
Comment (usamazansari) [reply libs/features/profile/src/lib/components/profile-edit/profile-edit.service.ts:13]: Resolved in https://github.com/usamazansari/expenses-tracker/pull/62/commits/1bf47dc47a6e3117892d00c9b2316ef2f40759cc

**Review by usama251993 — APPROVED** [review]
Looks good

File: libs/shared/common/src/lib/utilities/form.utils.ts (hunk 2 above)
Comment (usamazansari) [reply libs/shared/common/src/lib/utilities/form.utils.ts:11]: Resolved in https://github.com/usamazansari/expenses-tracker/pull/62/commits/23dcbcd6064df9bcf23c155f20386b21e0cb95f8

File: libs/shared/common/src/lib/utilities/form.utils.ts (hunk 4 above)
Comment (usamazansari) [reply libs/shared/common/src/lib/utilities/form.utils.ts:42]: Resolved in https://github.com/usamazansari/expenses-tracker/pull/62/commits/23dcbcd6064df9bcf23c155f20386b21e0cb95f8

File: libs/shared/common/src/lib/utilities/form.utils.ts (hunk 3 above)
Comment (usamazansari) [reply libs/shared/common/src/lib/utilities/form.utils.ts:21]: Resolved in https://github.com/usamazansari/expenses-tracker/pull/62/commits/23dcbcd6064df9bcf23c155f20386b21e0cb95f8

File: libs/shared/common/src/lib/utilities/string.utils.ts (hunk 5 above)
Comment (usamazansari) [reply libs/shared/common/src/lib/utilities/string.utils.ts:1]: Resolved in https://github.com/usamazansari/expenses-tracker/pull/62/commits/23dcbcd6064df9bcf23c155f20386b21e0cb95f8

**Pull request merged**